website/frontend/src/context/editor/Provider.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+import { useAlert } from 'react-alert';
+import JSZip from 'jszip';
+import saveAs from 'file-saver';
+import { EditorContext } from './Context';
+import { Props } from '../types';
+import { LOCAL_STORAGE, WASM_COMPILER_GET } from 'consts';
+
+const { Provider } = EditorContext;
+
+const useEditor = () => {
```

Comment:
this code looks like method or separate hook but not context i think

Reply:
Previously it was in the `Header` component. I agree, it's overall massive and awkward, hence worth refactoring.